tests/components/firmata/test_config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Test the Firmata config flow."""
+from pymata_express.pymata_express_serial import serial
+
+from homeassistant import config_entries, setup
+from homeassistant.components.firmata.const import CONF_SERIAL_PORT, DOMAIN
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+
+from tests.async_mock import patch
+
+
+async def test_import_cannot_connect_pymata(hass: HomeAssistant) -> None:
+    """Test we fail with an invalid board."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    with patch(
+        "homeassistant.components.firmata.board.PymataExpress.start_aio",
+        side_effect=RuntimeError,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={CONF_SERIAL_PORT: "/dev/nonExistent"},
+        )
+
+    assert result["type"] == "abort"
+    assert result["reason"] == "cannot_connect"
+    await hass.async_block_till_done()
```

Comment:
We don't need to block till done at the end of the test.

We use that method to wait until the current tasks are done. If we don't need to wait for the tasks we don't need to await this method.